lib/radiator/directory/editor.ex

```diff
@@ -421,6 +423,44 @@ defmodule Radiator.Directory.Editor do
     episode
   end
 
+  def get_chapter(actor = %Auth.User{}, id) do
+    case Repo.get(Chapter, id) do
+      nil ->
+        @not_found_match
+
+      chapter = %Chapter{} ->
+        if has_permission(actor, chapter, :readonly) do
+          {:ok, chapter}
+        else
+          @not_authorized_match
+        end
+    end
+  end
+
+  def create_chapter(actor = %Auth.User{}, audio, attrs) do
+    if has_permission(actor, audio, :edit) do
+      AudioMeta.create_chapter(audio, attrs)
+    else
+      @not_authorized_match
+    end
+  end
+
+  def update_chapter(actor = %Auth.User{}, chapter = %Chapter{}, attrs) do
+    if has_permission(actor, chapter, :edit) do
+      AudioMeta.update_chapter(chapter, attrs)
```

Comment:
update_chapter needs to delete the old image file if a new one is uploaded. otherwise we leak files in the store (just tried by updating some chapter)